File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.py
class Solution(object):
    def countPartitions(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        c=0
        for i in range(1,len(nums)):
            left_sum=sum(nums[:i])
            right_sum=sum(nums)-left_sum
            if left_sum%2==right_sum%2: c+=1
        return c

File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.cpp
class Solution {
public:
    int countPartitions(vector<int>& nums) {
        int c=0,i=0,total=0,right_sum;
        while (i<nums.size()) total+=nums[i++];
        for (int i=1; i<nums.size(); i++)
        {
            int left_sum=0,j=0;
            while (j<i) left_sum+=nums[j++];
            right_sum=total-left_sum;
            if (left_sum%2==right_sum%2) c+=1;
        }
        return c;
    }
};